lib/babel.js

```diff
@@ -74,8 +77,33 @@ if (inputSourceMap) {
 	options.inputSourceMap = JSON.parse(inputSourceMap.map);
 }
 
+// load up custom compilers
+process.argv.slice(3).forEach(function (arg) {
+	var match = /^--compilers=.+?:(.+)$/i.exec(arg);
+	if (match) {
+		require(match[1]);
+	}
+});
+
 // include test file
-var transpiled = babel.transformFileSync(testPath, options);
+var Module = module.constructor;
+var source = fs.readFileSync(testPath, 'utf8');
+var extension = path.extname(testPath);
+
+var compiler = Module._extensions[extension];
+if (compiler) {
+	compiler(
+		{
+			_compile: function (s, f) {
+				source = s;
+				testPath = f;
```

Comment:
Yep. This was quick and dirty. Final solution would need that. I didn't want to invest a lot of time in it, because I'm leery of mocking an undocumented API this way. To be honest, I anticipated a little more of an "Are you nuts?" reaction to this.

Do you see any need to mock more of the implementation? I have not yet seen an extension that uses anything beyond `_compile`.